dev/NumberBox/NumberBox.cpp

```diff
@@ -642,3 +668,19 @@ bool NumberBox::IsInBounds(double value)
     return (value >= Minimum() && value <= Maximum());
 }
 
+void NumberBox::UpdateTextBoxCornerRadiusForInlineMode()
+{
+    if (m_isControlCornerRadiusAvailable)
+    {
+        if (auto && textBox = m_textBox.get())
+        {
+            auto value = this->CornerRadius();
+
+            // Respect T-rule in inline mode. Since a spin button borders the textbox to the right
+            // we are not allowed to round the right corners of the textbox.
+            value.BottomRight = value.TopRight = 0;
+            textBox.CornerRadius(value);
```

Comment:
Ideally we wouldn't want to do it quite like this as if someone retemplates the control to put the inline buttons on the left side (like I have 😅) leaving the TextBox on the right, we're breaking the corners again.

Reply:
Yes, I agree. I had this thought as well. If we can put this T-rule enforcement in the control template itself (via a Visual State setter for example) that would be best (as then this could easily be modified to suit the needs of the control redesign).

Reply:
@JohnnyWestlake Ok, instead of enforcing the T-rule in code-behind now, I moved that code back into the control template:
```xml
<VisualState x:Name="SpinButtonsVisible">
    <VisualState.Setters>
        <Setter Target="DownSpinButton.Visibility" Value="Visible" />
        <Setter Target="UpSpinButton.Visibility" Value="Visible" />
        <contract7Present:Setter Target="InputBox.CornerRadius" Value="{Binding RelativeSource={RelativeSource TemplatedParent}, Path=CornerRadius, Converter={StaticResource LeftCornerRadiusFilterConverter}}" />
    </VisualState.Setters>
</VisualState>
```

In code-behind, I replaced setting the borer with a call to the *SpinButtonsVisible* visual state *every* time the Numberbox' CornerRadius is changed and we are in the *inline* SpinButtonPlacement mode. If you retemplated your numberbox to have the buttons on the left side of the numberbox, all the corner radius changes you have to do are now solely in the control temlate again.

Reply:
And here's a gif with a quick retemplate:
![numberbox_cornerradius_issue_retemplate](https://user-images.githubusercontent.com/1398851/78424135-a6c92600-766b-11ea-899b-43f47daa8594.gif)